src/cmd/hotspots/transfer.rs

```diff
@@ -0,0 +1,136 @@
+    #[structopt(name = "TRANSACTION")]
+    txn: Option<String>,
+    #[structopt(long)]
+    /// Commit to sign and submit the transaction
+    commit: bool,
+}
+
+impl Buy {
+    fn read_txn(&self) -> Result<String> {
+        match &self.txn {
+            Some(txn) => Ok(txn.to_string()),
+            None => {
+                let mut buffer = String::new();
+                io::stdin().read_line(&mut buffer)?;
+                Ok(buffer.trim().to_string())
+            }
+        }
+    }
+}
+
+impl Transfer {
+    pub fn run(self, opts: Opts) -> Result {
+        let wallet = load_wallet(opts.files)?;
+        let client = Client::new_with_base_url(api_url());
+
+        match self {
+            Self::Sell(sell) => {
+                let seller = wallet.address_as_vec();
+                let buyer = PubKeyBin::from_b58(&sell.buyer)?;
+                let buyer_account = client.get_account(&buyer.to_b58()?)?;
+                let gateway = PubKeyBin::from_b58(&sell.gateway)?.to_vec();
+
+                let mut txn = BlockchainTxnTransferHotspotV1 {
+                    fee: 0,
+                    seller,
+                    gateway,
+                    buyer: buyer.to_vec(),
+                    seller_signature: vec![],
+                    buyer_signature: vec![],
+                    amount_to_seller: if let Some(price) = sell.price {
+                        price.to_bones()
+                    } else {
+                        0
+                    },
+                    buyer_nonce: buyer_account.speculative_nonce + 1,
+                };
+                txn.fee = txn.txn_fee(&get_txn_fees(&client)?)?;
+                println!("{:#?}", txn.to_json()?);
+
+                let password = get_password(false)?;
+                let keypair = wallet.decrypt(password.as_bytes())?;
+                txn.seller_signature = txn.sign(&keypair)?;
+                println!("{}", txn.in_envelope().to_b64()?);
+            }
+
+            Self::Buy(buy) => {
+                let mut envelope = BlockchainTxn::from_b64(&buy.read_txn()?)?;
+
+                match &mut envelope.txn {
+                    Some(Txn::TransferHotspot(t)) => {
+                        // verify that nonce is still valid.
+                        let nonce = t.buyer_nonce;
+                        let buyer_account =
+                            client.get_account(&PubKeyBin::from_vec(&t.buyer).to_b58()?)?;
+                        let expected_nonce = buyer_account.speculative_nonce + 1;
+
+                        if buyer_account.speculative_nonce + 1 != nonce {
+                            println!(
+                                "Buyer_nonce in transaction is {} while expected nonce is {}",
+                                nonce, expected_nonce
+                            );
+                            return Err("Hotspot transfer nonce no longer valid".into());
+                        }
+
+                        println!("{:#?}", t.to_json()?);
```

Comment:
use print_json print_txn?

Reply:
I guess the print_json / print_txn pattern would be another reason to revert to the current `--commit` standard. Otherwise, I don't have the hash when I print the transaction might way.